File: apps/backend/src/EnterpriseAgentOs.Domain/features/Agents/Interfaces/IBrowserSessionRepository.cs
namespace EnterpriseAgentOs.Domain.Features.Agents;

public interface IBrowserSessionRepository
{
    Task<BrowserSessionRecord?> GetByAgentAsync(Guid agentId, CancellationToken ct = default);
    Task<BrowserSessionRecord> UpsertAsync(Guid agentId, string runtimeSessionId, string? cookiesJson, CancellationToken ct = default);
    Task DeleteByAgentAsync(Guid agentId, CancellationToken ct = default);
}

File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/features/agents/interfaces/IBrowserSessionRepository.java
package com.enterpriseagentos.backendjava.domain.features.agents.interfaces;

import java.util.UUID;
import java.util.concurrent.CompletableFuture;

import com.enterpriseagentos.backendjava.domain.features.agents.models.BrowserSessionModel;

public interface IBrowserSessionRepository {
    CompletableFuture<BrowserSessionModel> getByAgentAsync(UUID agentId);
    CompletableFuture<BrowserSessionModel> upsertAsync(UUID agentId, String runtimeSessionId, String cookiesJson);
    CompletableFuture<Void> deleteByAgentAsync(UUID agentId);
}
